### ISSUE TITLE
Unsupported operation Transpose?

### ISSUE THREAD

**Issue by d-lareg** [ARM-software/armnn#203]
Hello,

I am currently try to load my small model using the TF-Parser. The architecture is very simple: 
Conv2d + ReLU -> Conv2d + ReLU -> Flatten -> Dense -> Dense

My training environment is: 
OS: Win10
Framework: Keras of Tensorflow 1.13 
Device: CPU
Dataformat: NHWC

Inference environment:
ArmNN + ArmCL: 19.02

After training I converted the Keras model (.hdf5) into a .pb file, freezed the variables and removed the dropout layers. Unfortunately I have problems loading the model and get the following error: 

> Unsupported operation Transpose in tensorflow::GraphDef  at function LoadNodeDef [/armnn/armnn-19.02/src/armnnTfParser/TfParser.cpp:3237]

As far as i can see it is related to the flatten layer, what makes me wonder since the flatten layer is supported by ArmCL. Isn't it supported by ArmNN? What could cause this error?

Thanks in advance!

**Comment by d-lareg** [comment]
I tried to use the MNIST model from the MNIST example (simple_mnist_tf.pb) but without any success. I get the following error message:

> Failed to parse protobuf file /tmp/simple_mnist_tf.pb  at function CreateNetworkFromBinaryFile [/armnn/armnn-19.02/src/armnnTfParser/TfParser.cpp:3388]

**Comment by MatthewARM** [comment]
Hi @d-lareg we're a bit further ahead with TfLite compatibility than we are with Tensorflow. Perhaps you could try converting your model to .tflite format first? 

This would also give you the benefit that tflite_convert performs a standard set of optimisations on the graph, such as removing training-only nodes.

**Comment by d-lareg** [comment]
Hi @MatthewARM thanks for your advice! I will try that!
For the moment i changed the model to use only Conv2D layers. That fixed the Transpose error. 
I furthermore figured out that keras is doing some weird stuff. I used the activation param of the Conv2D class when defining the model, which turned out to be a failure. For example, by doing so Keras is implementing the Softmax actiation in a way that can not be read by the TF-Parser. Better use Conv2D + a seperate Activation layer, which works fine for the Softmax activation. Another Keras thing is the implementation of the Flatten layer. More informations can be found here: [OpenCV-Forum](https://github.com/opencv/opencv_contrib/issues/1241#issuecomment-312811689)

**Comment by MatthewARM** [comment]
Thanks @d-lareg that's useful information.
